frontend/src/utils/date.ts

```diff
@@ -140,3 +142,21 @@ export const transformShortIsoStringToDate = (
 export const transformDateToShortIsoString = (date: unknown): string | null => {
   return isDate(date) ? format(date as Date, 'yyyy-MM-dd') : null
 }
+
+/** Transforms the invalid days array to checkbox group value */
+export const transformInvalidDaysToCheckedBoxesValue = (
+  invalidDays: InvalidDaysOptions[],
+): InvalidDaysOptions[] => {
+  return Object.values(InvalidDaysOptions).filter(
```

Comment:
perf: should we optimize this function? 

```tsx
const ALL_INVALID_DAYS = new Set(Object.values(InvalidDaysOptions))

// inside function
export const transformInvalidDaysToCheckedBoxesValue = (
  invalidDays: InvalidDaysOptions[],
): InvalidDaysOptions[] => {
 return invalidDays.filter(day => !ALL_INVALID_DAYS.has(value));
}
```

Reply:
oh let me rewrite that real quick this grabs the difference from the wrong array lol

Reply:
Can I check how this is more optimised compared to the one that I wrote? Is it because of `Object.values(InvalidDaysOptions)`?

Reply:
`Array.prototype.includes` is always gonna be slower than `Set.prototype.has`

Reply:
Oh I didn't know `includes` is an O(n) operation. Thanks for the suggestion!